tests/PrepareRename.test.cpp

```diff
@@ -0,0 +1,193 @@
+#include "doctest.h"
+#include "Fixture.h"
+
+#include "LSP/IostreamHelpers.hpp"
+
+TEST_SUITE_BEGIN("PrepareRename");
+
+TEST_CASE_FIXTURE(Fixture, "return_nullopt_on_nothing_to_rename")
+{
+    auto uri = newDocument("foo.luau", "");
+
+    lsp::PrepareRenameParams params = {lsp::TextDocumentIdentifier{uri}, lsp::Position{0, 0}};
+    auto result = workspace.prepareRename(params);
+    REQUIRE(result == std::nullopt);
+}
+
+TEST_CASE_FIXTURE(Fixture, "handle_generic_type_parameter")
+{
+    auto source = R"(
+        type Ty<S> = S
+    )";
+
+    auto uri = newDocument("foo.luau", source);
+
+    lsp::PrepareRenameParams params = {lsp::TextDocumentIdentifier{uri}, lsp::Position{1, 16}};
+    auto result = workspace.prepareRename(params);
+    REQUIRE(result);
+    REQUIRE(result.value().placeholder == "S");
+    REQUIRE(result.value().range == lsp::Range{{1, 16}, {1, 17}});
+}
+
+TEST_CASE_FIXTURE(Fixture, "handle_generic_type_parameter_reference")
+{
+    auto source = R"(
+        type Ty<S> = S
+    )";
+
+    auto uri = newDocument("foo.luau", source);
+
+    lsp::PrepareRenameParams params = {lsp::TextDocumentIdentifier{uri}, lsp::Position{1, 21}};
+    auto result = workspace.prepareRename(params);
+    REQUIRE(result);
+    REQUIRE(result.value().placeholder == "S");
+    REQUIRE(result.value().range == lsp::Range{{1, 21}, {1, 22}});
+}
+
+TEST_CASE_FIXTURE(Fixture, "handle_generic_type_pack_parameter")
+{
+    auto source = R"(
+        type Ty<S...> = S... & { ... }
+    )";
+
+    auto uri = newDocument("foo.luau", source);
+
+    lsp::PrepareRenameParams params = {lsp::TextDocumentIdentifier{uri}, lsp::Position{1, 16}};
+    auto result = workspace.prepareRename(params);
+    REQUIRE(result);
+    REQUIRE(result.value().placeholder == "S");
+    REQUIRE(result.value().range == lsp::Range{{1, 16}, {1, 17}});
+}
+
+TEST_CASE_FIXTURE(Fixture, "handle_generic_type_pack_parameter_reference")
+{
+    auto source = R"(
+        type Ty<S...> = S... & { ... }
+    )";
+
+    auto uri = newDocument("foo.luau", source);
+
+    lsp::PrepareRenameParams params = {lsp::TextDocumentIdentifier{uri}, lsp::Position{1, 24}};
+    auto result = workspace.prepareRename(params);
+    REQUIRE(result);
+    REQUIRE(result.value().placeholder == "S");
+    REQUIRE(result.value().range == lsp::Range{{1, 24}, {1, 25}});
+}
+
+TEST_CASE_FIXTURE(Fixture, "handle_type_prefix")
+{
+    auto source = R"(
+        local Module = require("")
+        type Ty = Module.Ty
+    )";
```

Comment:
Lets add a test for type names too? i.e. the `Ty` in this spot (both prefixed and unprefixed)